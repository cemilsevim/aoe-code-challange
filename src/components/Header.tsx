import React from "react";
import { Button, Form, FormControl, Nav, Navbar, NavDropdown, Container } from "react-bootstrap";
import {
    Link,
} from "react-router-dom";

function Header() {
    return (
        <Navbar bg="light" expand="lg">
            <Container>
                <Navbar.Brand href="#home">Age of Empires</Navbar.Brand>
                <Navbar.Toggle aria-controls="basic-navbar-nav" />
                <Navbar.Collapse id="basic-navbar-nav">
                    <Nav className="mr-auto">
                        <Nav.Item>
                            <Link className="nav-link" to="/">Home</Link>
                        </Nav.Item>
                        <Nav.Item>
                            <Link className="nav-link" to="/units">Units</Link>
                        </Nav.Item>
                    </Nav>
                </Navbar.Collapse>
            </Container>
        </Navbar>
    );
}

export default Header;