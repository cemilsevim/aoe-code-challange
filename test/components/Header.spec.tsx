import renderer from 'react-test-renderer';
import Header from '../../src/components/Header';
import {
    BrowserRouter
} from "react-router-dom";
import ReactDOM from "react-dom";

describe('Header', () => {
    it('renders correctly', () => {
        const header = (
            <BrowserRouter>
                <Header />
            </BrowserRouter>
        );
        ReactDOM.render(header, document.createElement('div'));
        
        const headerRenderer = renderer.create(header);
        expect(headerRenderer.toJSON()).toMatchSnapshot();  
    });
});