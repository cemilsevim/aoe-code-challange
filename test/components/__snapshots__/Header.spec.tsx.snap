// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      Age of Empires
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <div
          className="nav-item"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
        </div>
        <div
          className="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link"
            href="/units"
            onClick={[Function]}
          >
            Units
          </a>
        </div>
      </div>
    </div>
  </div>
</nav>
`;
